### ISSUE TITLE
|| operator: Not failglob sensitive

### ISSUE THREAD

**Issue by anordal** [fish-shell/fish-shell#8196]
To my surprise, this `exit` is unreachable:

    echo nonexist/* || exit $status

But thankfully, this works

    echo nonexist/*; or exit $status

as does this:

    if echo nonexist/*
        true
    else
        exit $status
    end

I noticed this in fish 3.2.2, but I also tested the latest commit (06acc201), and it's still the case.

    > sh -c 'env HOME=$(mktemp -d) fish'
    > printf '%s\n' $version (fish --version) (rpm -q fish) (lsb-release -sd)
    3.2.2
    fish, version 3.2.2
    fish-3.2.2-1.2.x86_64
    "openSUSE Tumbleweed"

I think this is a minor issue, since the workaround is easy. Just surprising.

Now for some entertainment: The same in Bash – failglob without errexit – is even more inconsistent and defying of sensible ways to use it:

```bash
#!/usr/bin/env bash
shopt -s failglob

echo nonexist/* || exit $?; echo Unreachable
echo Failed, but still here: $?

echo nonexist/*; echo Unreachable
echo Failed and still here: $?

if echo nonexist/*; then
    echo Unreachable
else
    echo Unreachable
    exit $?
fi; echo Unreachable
echo Failed, but still here: $?
```

**Comment by faho** [comment]
Yeah, the `||` is still part of the same conjunction, and that's parsed as one bit, and then the `*` fails the entire thing. That's probably wrong.

However, since the error message here is unsilenceable, it's not high priority - the simple answer is to not do that.

If you want to have a glob that can fail, use it with `for`, `set`, `count` or a `VAR=VAL` override.
